openmdao/recorders/sqlite_recorder.py

```diff
@@ -361,6 +363,7 @@ def startup(self, recording_requester):
             var_settings.update(objectives)
             var_settings.update(constraints)
             var_settings = self._cleanup_var_settings(var_settings)
+            var_settings['execution_order'] = var_order
```

Comment:
Should we update the case version for this change, and add a backwards-compatibility test?

Reply:
We have not been changing the version when the content of the metadata changes.. only when the data format changes.
The presence of the data field is not required, and it will fall back to a sorted order.  I'll add a test for that branch of the code.

Reply:
added backwards-compatibility test